### ISSUE TITLE
No parallel downloads from alternate sources before restart

### ISSUE THREAD

**Issue by virtual-machinist** [airdcpp-web/airdcpp-webclient#350]
### Current UI and client versions:

Application version: AirDC++w 2.7.0 x86_64
Web UI version: 2.7.0

### Operating system:

Ubuntu 18.04.4 LTS

### Steps to reproduce the issue:

Add a directory from Search with multiple sources available. Bundle is downloaded only from the first source until the client is restarted regardless of current bandwidth utilization (I do have _no new downloads if bandwidth goes above X KiB/s_ option set, but it's nowhere near the set limit). If the original source is removed the download stalls until the client is restarted.

Don't know if this helps but the setup I run has most configuration settings adjusted manually, i.e. active mode, external IP, bandwidth limits, slots, etc. I cannot allow automatic configuration because the instance is firewalled and shares connection with other bandwidth-heavy applications.

**Comment by maksis** [comment]
Do I understand that all sources are being added for the bundle but the other ones don't appear in the transfer view at all?

**Comment by virtual-machinist** [comment]
Yes, exactly. I do see them when I select "Manage files" and they are offline.

**Comment by maksis** [comment]
That's strange, as users should be added in the transfer view with an error even if the download can't be started. Possibly it's related to your config as I can't reproduce the issue. 

Are you able test it with the default config (excluding connectivity settings, as those shouldn't really matter)? Better also try with the latest beta (https://web-builds.airdcpp.net/develop/), just to be sure. Possibly you could also send your config file (`DCPlusPlus.xml`) to me, e.g. via the dev hub (adcs://web-dev.airdcpp.net:1511).

**Comment by virtual-machinist** [comment]
OK. I'll try the latest beta sometime later, see if it helps. Added the XML config here with all non-essential parts (nick, IP, ports, search history) redacted.
[DCPlusPlus.xml.txt](https://github.com/airdcpp-web/airdcpp-webclient/files/4417101/DCPlusPlus.xml.txt)

**Comment by maksis** [comment]
Thanks. Does this happen with ADC users (grouped directories with multiple users)? Does it happen with files too?

**Comment by maksis** [comment]
Oh, and which client your sources are using (AirDC++ or something else)? The config looks quite normal and I'm not able to reproduce the issue with it.

**Comment by maksis** [comment]
Is this issue still happening?

**Comment by virtual-machinist** [comment]
I am afraid I cannot look into this issue further as the machine I was running the client on is offline for the time being. Regardless if this helps - the issue was present with files from any peer I could connect to (i.e. be it vanilla DC++, ApexDC++, AirDC++, ...).

My use case was a couple of private plain non-ADC hubs. Sources for files were found and matched, but were never downloaded from other peers except the first one until I restarted the client. I never tried ADC as no hubs I connect to use it.

Unfortunately I could not upgrade to the latest beta before the machine went offline. If the issue cannot be replicated with what is written here, I think this ticket can be closed for the time being. I don't think I'll be able to provide any input until I get the client running again (unfortunately not earlier than late autumn or even winter).

**Comment by virtual-machinist** [comment]
Can confirm this issue on the latest development version (downloaded [here](https://web-builds.airdcpp.net/develop/)). I've kept the settings intact, i.e. restored them from the previous version. Operating system has been upgraded to Xubuntu 20.04.1 LTS.

**Comment by RyanRadly** [comment]
This is still an issue on 2.11.1 (Web UI also 2.11.1).  Adding a full directory from search defaults to the original singular source initially.  I can get it to resolve by doing a reconnect to the Hub instead of a full on restart of the program.  Operating system is Ubu 18.04.

**Comment by maksis** [comment]
If someone is willing to help with investigating the issue, please join the dev hub at adcs://web-dev.airdcpp.net:1511

It might help if I could access a web UI that can be used to reproduce this issue.

**Comment by maksis** [comment]
This is most likely fixed in 2.12.0 beta (https://github.com/airdcpp-web/airdcpp-webclient/issues/350). Can anyone confirm this?

**Comment by Kcchouette** [comment]
For me it was somewhat related to #447
